HDOS-437

pka1024 opened the pull request:
feat: (api):
- Workflow-editor renders flowchart from workflowTransformation.

Commit:
API change, flowchart-converter

feat: (api):
- Flowchart-converter uses now transformations, in progress.

Commit:
API change, flowchart-converter

feat: (api):
- Workflow-editor renders flowchart from workflowTransformation.

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -461,7 +464,7 @@ export class WorkflowEditorComponent {
       operator.name = data;
       this.workflowService.updateWorkflow(this.currentWorkflow);
-      this._convertWorkflowToFlowchart(this.currentWorkflow);
+      // this._convertWorkflowToFlowchart(this.currentWorkflow);
```
TODO

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -67,70 +66,88 @@ export class FlowchartConverterService {
-
-    this.convertWorkflowConstants(workflowClean, flowchart);
+    // TODO no need anymore
+    // this.convertWorkflowConstants(workflowClean, flowchart);
```
wieder einbauen

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -144,39 +161,41 @@ export class FlowchartConverterService {
     const workflowIO: Array<FlowchartComponentIO> = [];
-    for (const io of workflow.inputs) {
+    for (const io of workflow.content.inputs) {
+      // TODO No constants
```
löschen

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -144,39 +161,41 @@ export class FlowchartConverterService {
 
-    for (const io of workflow.outputs) {
+    for (const io of workflow.content.outputs) {
+      // TODO No constants
```
löschen

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -190,16 +209,42 @@ export class FlowchartConverterService {
-      const linkIds = link.path.map(point => point.id);
+
+    for (const link of workflow.content.links) {
+      // filter constant links
```
löschen

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -190,16 +209,42 @@ export class FlowchartConverterService {
+      }
+
+      const linkPath = link.path.map(position => [position.x, position.y]);
+      // TODO Check for id
```
löschen

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -190,16 +209,42 @@ export class FlowchartConverterService {
+      // const linkIds = link.path.map(point => point.id);
+      const linkIds = link.path.map(() => UUID().toString());
+
+      let linkStartOperator = '';
```
Kommentar:
Links which lead to an input / output of the workflow itself do not have a start / end operator id (since the operator is not part of the workflow content but it is the workflow itself). The flowchart however needs an operator id for every link, that is why we use the id of the workflow.

const linkStartOperator = link.start.operator ?? workflow.id;
const linkEndOperator = ...

Commit:
API change, flowchart-converter

feat: (api):
- Removed some TODO comments.
- Reactivated the convertWorkflowConstants method and deleted the other constant filter.
- Added a comment why some linkStartOperator and linkEndOperator, use the id of the workflow.

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -74,43 +84,47 @@ export class FlowchartConverterService {
-      connector.value = io.constantValue.value;
+      connector.value = constant.value;
 
       flowchart.links = flowchart.links.filter(
```
// TODO: Maybe links to constants can be removed from the API completely, since they are deleted here anyway.

Commit:
API change, flowchart-converter

feat: (api):
- Added a TODO comment.
